deviceadvisor/tests/mqtt_publish/main.cpp

```diff
@@ -0,0 +1,527 @@
+
+        if (!s_cmdOptionExists(argv, argv + argc, "--x509_thing"))
+        {
+            fprintf(stdout, "X509 credentials sourcing requires an x509 thing name to be specified.\n");
+            s_printHelp();
+            return 1;
+        }
+        x509ThingName = s_getCmdOption(argv, argv + argc, "--x509_thing");
+
+        if (!s_cmdOptionExists(argv, argv + argc, "--x509_cert"))
+        {
+            fprintf(stdout, "X509 credentials sourcing requires an Iot thing certificate to be specified.\n");
+            s_printHelp();
+            return 1;
+        }
+        x509CertificatePath = s_getCmdOption(argv, argv + argc, "--x509_cert");
+
+        if (!s_cmdOptionExists(argv, argv + argc, "--x509_key"))
+        {
+            fprintf(stdout, "X509 credentials sourcing requires an Iot thing private key to be specified.\n");
+            s_printHelp();
+            return 1;
+        }
+        x509KeyPath = s_getCmdOption(argv, argv + argc, "--x509_key");
+
+        if (s_cmdOptionExists(argv, argv + argc, "--x509_rootca"))
+        {
+            x509RootCAFile = s_getCmdOption(argv, argv + argc, "--x509_rootca");
+        }
+
+        useX509 = true;
+    }
+
+    if (s_cmdOptionExists(argv, argv + argc, "--message"))
+    {
+        messagePayload = s_getCmdOption(argv, argv + argc, "--message");
+    }
+
+    if (s_cmdOptionExists(argv, argv + argc, "--count"))
+    {
+        int count = atoi(s_getCmdOption(argv, argv + argc, "--count"));
+        if (count > 0)
+        {
+            messageCount = count;
+        }
+    }
+
+    /********************** Now Setup an Mqtt Client ******************/
+    /*
+     * You need an event loop group to process IO events.
+     * If you only have a few connections, 1 thread is ideal
+     */
+    Io::EventLoopGroup eventLoopGroup(1);
+    if (!eventLoopGroup)
+    {
+        fprintf(
+            stderr, "Event Loop Group Creation failed with error %s\n", ErrorDebugString(eventLoopGroup.LastError()));
+        exit(-1);
+    }
+
+    Aws::Crt::Io::DefaultHostResolver defaultHostResolver(eventLoopGroup, 1, 5);
+    Io::ClientBootstrap bootstrap(eventLoopGroup, defaultHostResolver);
+
+    if (!bootstrap)
+    {
+        fprintf(stderr, "ClientBootstrap failed with error %s\n", ErrorDebugString(bootstrap.LastError()));
+        exit(-1);
+    }
+
+    Aws::Crt::Http::HttpClientConnectionProxyOptions proxyOptions;
+    if (!proxyHost.empty())
+    {
+        proxyOptions.HostName = proxyHost;
+        proxyOptions.Port = proxyPort;
+        proxyOptions.AuthType = Aws::Crt::Http::AwsHttpProxyAuthenticationType::None;
+    }
+
+    Aws::Crt::Io::TlsContext x509TlsCtx;
+    Aws::Iot::MqttClientConnectionConfigBuilder builder;
+
+    if (!certificatePath.empty() && !keyPath.empty())
+    {
+        builder = Aws::Iot::MqttClientConnectionConfigBuilder(certificatePath.c_str(), keyPath.c_str());
+    }
+    else if (useWebSocket)
+    {
```

Comment:
Ditto for this and the subscribe - if it's not needed, I'd remove the websocket stuff.